ext/spl/spl_directory.c

```diff
@@ -693,11 +693,8 @@ zend_function *spl_filesystem_object_get_method_check(zend_object **object, zend
 	spl_filesystem_object *fsobj = spl_filesystem_from_obj(*object);
 
 	if (fsobj->u.dir.dirp == NULL && fsobj->orig_path == NULL) {
-		zend_function *func;
-		zend_string *tmp = zend_string_init("_bad_state_ex", sizeof("_bad_state_ex") - 1, 0);
-		func = zend_std_get_method(object, tmp, NULL);
-		zend_string_release_ex(tmp, 0);
-		return func;
+		zend_throw_error(NULL, "The parent constructor was not called: the object is in an invalid state");
```

Comment:
We'll probably want to move this check into each method and drop the get_method overload. That way the stack trace should be correct. Can you adjust the test to also check stack trace?

Reply:
So I tried to do this, but because userland can extend the class and rewrite the method, this check will not be done if moved into each function, that's why I think it's a get handler and I don't know how to fix it.

Reply:
Isn't the method final?

Reply:
This method is final, but I was talking about how *other* methods can be overwritten and thus bypass a check we could introduce in the method to see if the object is properly initialised. And I think the get_method overload checks that a userland class can't put the object in an invalid state (which mainly seems to be an issue for the GlobIterator)